Automattic/jetpack
# replace is_active to determine available xmlrpc methods

Stop using is_active to determine what XMLRPC endpoints should be available for the requests.

This is part of the cleanup to slowly deprecate the `is_active` method and open the path to the user-less project.

#### Changes proposed in this Pull Request:

* Changes the criteria used to register XML-RPC methods

#### Jetpack product discussion


1191179647901802-as-1200087466875449

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






In order to test this PR, there is this handy script -> 28d59-pb/#php

It will make 3 request to `system.listMethods` in your site. Save this script in your sandbox `public_html` folder and edit it to add your test site ID.

* One with an unauthenticated request
* One authenticated with a blog token
* One authenticated with a user token

So the test flow is:

* Disconnect your site
* Run the script on your sandbox
* Connect only at a site level
* Run the script
* Connect  a user
* Run the script

This will give you 3 outputs. Each output with the three types of request.

Look at the table here -> p9dueE-1Wm-p2

Make sure the exposed methods match the table. Each of the script runs will represent one column in the table.

Also, please read the code, follow the logic and check the criteria are not altered.

Finally, make sure the Jetpack Debugger passes all the tests and test the connection flows.

**NOTE** For the Debugger tests to pass you'll need https://github.com/Automattic/jetpack/pull/19301 and D59309-code (both deployed/merged already)
